x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -242,19 +246,20 @@ protected void masterOperation(
                     client.execute(RollupIndexerAction.INSTANCE, rollupIndexerRequest, ActionListener.wrap(indexerResp -> {
                         if (indexerResp.isCreated()) {
                             // 4. Make rollup index read-only and set the correct number of replicas
-                            final Settings settings = Settings.builder()
-                                .put(IndexMetadata.SETTING_BLOCKS_WRITE, true)
-                                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, sourceIndexMetadata.getNumberOfReplicas())
-                                .build();
-                            UpdateSettingsRequest updateSettingsReq = new UpdateSettingsRequest(settings, rollupIndexName);
+                            final Settings.Builder settings = Settings.builder().put(IndexMetadata.SETTING_BLOCKS_WRITE, true);
+                            // Number of replicas had been previously set to 0 to speed up index population
+                            if (sourceIndexMetadata.getNumberOfReplicas() > 0) {
+                                settings.put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, sourceIndexMetadata.getNumberOfReplicas());
+                            }
```

Comment:
I don't understand this piece, why do we copy replicas only if they are greater than 0? How would this behave if someone had something like `index.auto_expand_replicas: 0-2` where the number of replicas is variable? Shouldn't we be consistent and maintain the number of replicas as defined?

Reply:
The idea behind this is that when we create the rollup index we explicitly [set that `num_of_replicas=0`](https://github.com/elastic/elasticsearch/blob/52cf0f7de4d941a8e68d8fdb413fcf23334e9e2c/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java#L508). We do so, because we don't want the index to replicate while we populate the index.

When all documents have been written in the index, we set the num_of_replicas equal to the num_of_replicas of the source index. Of course,  if num_of_replicas of the source index is 0, we don't have to do anything, hence the conditional.